src/db/spanner/models.rs

```diff
@@ -1153,19 +1158,34 @@ impl SpannerDb {
         let mut inserts = vec![];
         let mut updates = HashMap::new();
         let mut success = vec![];
+        let mut load_size: usize = 0;
         for bso in params.bsos {
             success.push(bso.id.clone());
             if existing.contains(&bso.id) {
                 let (columns, values) = bso_to_update_row(&user_id, collection_id, bso, timestamp)?;
+                load_size += values.compute_size() as usize;
                 updates
                     .entry(columns)
                     .or_insert_with(|| vec![])
                     .push(values);
             } else {
                 let values = bso_to_insert_row(&user_id, collection_id, bso, timestamp)?;
+                load_size += values.compute_size() as usize;
                 inserts.push(values);
             }
         }
+        if load_size > MAX_SPANNER_LOAD_SIZE {
+            self.metrics.clone().incr("sync.error.tooMuchData");
```

Comment:
Almost forgot this (and it's probably a separate issue). Using the "sync" prefix here, but I've seen "syncstorage" elsewhere. Probably want to standardize on "syncstorage" like the python version uses

Reply:
Agreed. Thanks!

Reply:
(Also going to normalize the `timer.*` var calls as `metrics.*` to make them easier to find in the sources.)